core/src/main/scala/besom/internal/Config.scala

```diff
@@ -168,6 +124,25 @@ object Config:
       else projectName
     Config(projectName, isProjectName = true)
 
+  extension (output: Output[Config])
+    def get[A : ConfigValueReader](key: NonEmptyString)(using Context): Output[Option[A]] =
+      output.flatMap(_.get[A](key))
+    def require[A : ConfigValueReader](key: NonEmptyString)(using Context): Output[A] =
+      output.flatMap(_.require[A](key))
+
+    def getString(key: NonEmptyString)(using Context): Output[Option[String]] = output.flatMap(_.getString(key))
+    def requireString(key: NonEmptyString)(using Context): Output[String] = output.flatMap(_.requireString(key))
+
+    def getDouble(key: NonEmptyString)(using Context): Output[Option[Double]] = output.flatMap(_.getDouble(key))
+    def requireDouble(key: NonEmptyString)(using Context): Output[Double] = output.flatMap(_.requireDouble(key))
+    
+    def getInt(key: NonEmptyString)(using Context): Output[Option[Int]] = output.flatMap(_.getInt(key))
+    def requireInt(key: NonEmptyString)(using Context): Output[Int] = output.flatMap(_.requireInt(key))
+
+    def getBoolean(key: NonEmptyString)(using Context): Output[Option[Boolean]] = output.flatMap(_.getBoolean(key))
+    def requireBoolean(key: NonEmptyString)(using Context): Output[Boolean] = output.flatMap(_.requireBoolean(key))
```

Comment:
Should we also supports maps and lists now?

Reply:
I think [this might be what you are referring to](https://www.pulumi.com/docs/concepts/config/#structured-configuration), and yes, we'd like to support this in the future.

Reply:
Structured configuration could also be available to besom users as a case class. That would require to make sure that `ConfigValueReader` is available to users so that they can implement an instance for a desired class. In a perfect case this should be reducible to adding `derives ConfigValueReader` (or whatever we decide to call this trait finally)

Reply:
this is probably [strongly typed config](https://www.pulumi.com/docs/concepts/config/#strongly-typed-configuration)

we have both on our roadmap: #207 #208